cmd/sbctl/main.go

```diff
@@ -307,10 +307,12 @@ func completionFishCmd() *cobra.Command {
 
 func main() {
 	rootCmd.PersistentPreRun = func(c *cobra.Command, args []string) {
+		fields := strings.Fields(c.CommandPath())
 		if strings.Contains(c.CommandPath(), "completion zsh") ||
 			strings.Contains(c.CommandPath(), "completion bash") ||
 			strings.Contains(c.CommandPath(), "completion fish") ||
-			strings.Contains(c.CommandPath(), "__complete") {
+			strings.Contains(c.CommandPath(), "__complete") ||
+			len(fields) > 1 && fields[1] == "help" {
```

Comment:
I wanted to use `Fields` to protect against the case where a user has `help` in some path passed to the other commands. If that isn't necessary, we should be fine using `Contains` directly.

`cobra` has `c.Name()`, but that gets hairy, since the completion commands are actually subcommands, so it initially returns the shell name.

Reply:
I'm wondering if we should swap the logic around. Instead of `PersistentPreRun` we can do a `PreRun` on all commands that require root. This can be a fairly simple:

```
for _, cmd := range []interface{createKeysCmd(), enrollKeysCmd()}{
    cmd.Prerun = func(....iforget) {
           // if check
    }
    rootCmd.AddCommand(cmd)
}
```

Add an variable for the commands or something. Wouldn't this be simpler?

Reply:
It has a chance of being much cleaner. I will see if I can implement it.